### ISSUE TITLE
Account Permission Re-direction Notification Adjustment

### ISSUE THREAD

**Issue by rsparker-utah** [uic-utah/uic-inventory#322]
I came across this discrepancy during testing of #143. I used my admin account to navigate to the site details, site contacts, and site location pages of a site (i.e., 124) not created by either my admin account or my public account. I then copied the following links for each:

- "Site Details" page link (https://uic-inventory.dev.utah.gov/site/124/add-details)
- "Site Contacts" page link (https://uic-inventory.dev.utah.gov/site/124/add-contacts)
- "Site Location" page link (https://uic-inventory.dev.utah.gov/site/124/add-location)

and pasted them into a separate browser logged into my public account and hit enter. Instead of being directed to the "How did you get here?" page, my public account was routed to the incomplete site notification page:
![image](https://github.com/uic-utah/uic-inventory/assets/49735267/c11686a0-c88a-4951-b34a-3134d67c2b64)

Would it be possible to make it so in cases like this for the public account where it is showing the incomplete inventory notification when navigating to links that the account shouldn't have access to (i.e., the site details, contacts, and location for sites the account didn't create), it instead showed the "How did you get here?" page? for example:
![image](https://github.com/uic-utah/uic-inventory/assets/49735267/31fda191-12c4-43ac-80c6-c0f63fd50459)

**Comment by steveoh** [comment]
I noticed that links to deleted inventories don't provide a good experience.

**Comment by steveoh** [comment]
The site is currently building and I believe I have addressed the original issue. Please validate.

**Comment by rsparker-utah** [comment]
This looks great and is working as expected. For reference, to test this, I used my admin account to navigate to the Site, details, contacts, and location pages for several sites not created by my public account (i.e., sites 124, 144, and 145). I then copied the links for each of these pages, logged into my public account in a separate browser, and tried to navigate to each of the copied links. For each of the links I received the updated notification:
![image](https://github.com/uic-utah/uic-inventory/assets/49735267/a9b2729b-8304-4473-8d1f-343c6deeb305)
